paimon-python/pypaimon/deletionvectors/bitmap_deletion_vector.py

```diff
@@ -0,0 +1,165 @@
+        return len(self._bitmap) == 0
+
+    def get_cardinality(self) -> int:
+        """
+        Returns the number of distinct integers added to the DeletionVector.
+
+        Returns:
+            The number of deleted positions.
+        """
+        return len(self._bitmap)
+
+    def merge(self, deletion_vector: DeletionVector) -> None:
+        """
+        Merge another DeletionVector to this current one.
+
+        Args:
+            deletion_vector: The other DeletionVector to merge.
+        """
+        if isinstance(deletion_vector, BitmapDeletionVector):
+            self._bitmap |= deletion_vector._bitmap
+        else:
+            raise RuntimeError("Only instance with the same class type can be merged.")
+
+    def serialize(self) -> bytes:
+        """
+        Serializes the deletion vector to bytes.
+
+        Returns:
+            The serialized bytes.
+        """
+        # Serialize the bitmap
+        bitmap_bytes = self._bitmap.serialize()
+
+        # Create the full data with magic number
+        magic_bytes = struct.pack('>I', self.MAGIC_NUMBER)
+        data = magic_bytes + bitmap_bytes
+
+        # Calculate size and checksum
+        size = len(data)
+        checksum = self._calculate_checksum(data)
+
+        # Pack: size (4 bytes) + data + checksum (4 bytes)
+        result = struct.pack('>I', size) + data + struct.pack('>I', checksum)
+
+        return result
+
+    @staticmethod
+    def deserialize_from_bytes(data: bytes) -> 'BitmapDeletionVector':
+        """
+        Deserializes a BitmapDeletionVector from bytes.
+
+        Args:
+            data: The serialized bytes (without magic number).
+
+        Returns:
+            A BitmapDeletionVector instance.
+        """
+        bitmap = BitMap.deserialize(data)
+        return BitmapDeletionVector(bitmap)
+
+    def bit_map(self):
+        return self._bitmap
+
+    def _check_position(self, position: int) -> None:
+        """
+        Checks if the position is valid.
+
+        Args:
+            position: The position to check.
+
+        Raises:
+            ValueError: If the position exceeds the maximum value.
+        """
+        if position > self.MAX_VALUE:
+            raise ValueError(
+                f"The file has too many rows, RoaringBitmap32 only supports files "
+                f"with row count not exceeding {self.MAX_VALUE}."
+            )
+
+    @staticmethod
+    def _calculate_checksum(data: bytes) -> int:
+        """
+        Calculates CRC32 checksum for the given data.
+
+        Args:
+            data: The data to calculate checksum for.
+
+        Returns:
+            The CRC32 checksum as an unsigned 32-bit integer.
+        """
+        return zlib.crc32(data) & 0xffffffff
+
+    def __eq__(self, other):
+        if not isinstance(other, BitmapDeletionVector):
+            return False
+        return self._bitmap == other._bitmap
+
+    def __hash__(self):
+        return hash(tuple(self._bitmap))
```

Comment:
BitmapDeletionVector.hash: Returns hash(tuple(self._bitmap)). This is inefficient on large bitmaps as it iterates through all bits, and it's rarely used. Consider removing the hash function or replacing it with one based on CRC/hash of serialized bytes (if a hash is truly needed).